pkg/genericapiserver/api/handlers/responsewriters/status.go

```diff
@@ -49,20 +48,14 @@ func apiStatus(err error) *metav1.Status {
 		//TODO: check for invalid responses
 		return &status
 	default:
-		status := http.StatusInternalServerError
-		switch {
-		//TODO: replace me with NewConflictErr
-		case storage.IsTestFailed(err):
```

Comment:
there is `storeerr.InterpretUpdateError` and it is also used it seems. But, we cannot make sure it is left out anymore easily: errors are passed around although these could probably be staticly typed. So it's not worth it to spend much time here, espcially because the dependency does not harm if we move pkg/storage.